### ISSUE TITLE
Do the functioning bbc-a11y tests return correct results?

### ISSUE THREAD

**Issue by EmmaJP** [bbc/bbc-a11y#70]
Catharine just raised the following ticket with the BBC framesworks team:

> The new bbc-a11y tool reports the following accessibility issues with ORB:
> 
> http://www.bbc.co.uk/news
> Field has no label or title attribute: /html/body/div/header/div/div[1]/div[3]/form/div/input[1]
> Field has no label or title attribute: /html/body/div/header/div/div[1]/div[3]/form/div/input[3]
> Field has no label or title attribute: /html/body/div/header/div/div[1]/div[3]/form/div/input[4]
> Form has no submit button: /html/body/div/header/div/div[1]/div[3]/form
> 
> http://www.bbc.co.uk/mundo
> Field has no label or title attribute: /html/body/div/header/div/div[1]/div[3]/form/div/input[1]
> Field has no label or title attribute: /html/body/div/header/div/div[1]/div[3]/form/div/input[2]
> Field has no label or title attribute: /html/body/div/header/div/div[1]/div[3]/form/div/input[4]
> Field has no label or title attribute: /html/body/div/header/div/div[1]/div[3]/form/div/input[5]
> Form has no submit button: /html/body/div/header/div/div[1]/div[3]/form
> 
> Theses issues are currently found for the News and Programmes team who runs the tool, please could you fix them.
> Copy and paste the xpath from the errors above into the console in the dev tools in Chrome in the following format then right click 'Reveal in Elements Panel' to show in code inspector:
> $x('/html/body/div/header/div/div[1]/div[3]/form/div/input[1]')

Looking more closely at the relevant guidelines (linked below) and the code specified in the errors (copied below), it seems the code does comply to the guidelines.  This implies that the bbc-a11y checks are returning incorrect errors.

http://www.bbc.co.uk/guidelines/futuremedia/accessibility/html/form-labels.shtml
http://www.bbc.co.uk/guidelines/futuremedia/accessibility/html/form-interaction.shtml

``` html
<form class="b-f" id="orb-search-form" role="search" method="get" action="http://search.bbc.co.uk/search" accept-charset="utf-8">
 <div>
   <label for="orb-search-q">Search the BBC</label>
   <input id="orb-search-q" type="text" name="q" placeholder="Search">
   <input type="image" id="orb-search-button" src="http://static.bbci.co.uk/frameworks/barlesque/3.2.4/orb/4/img/orb-search-dark.png" width="17" height="17" alt="Search the BBC">
    <input type="hidden" name="suggid" id="orb-search-suggid" value="">
 </div>
</form>
```

**Comment by greenc05** [comment]
With the latest version (15) of the tool, I only see the following errors now, was using version 0.0.13 before:

```
http://www.bbc.co.uk/news
Field has no label or title attribute: /html/body/div/header/div/div[1]/div[3]/form/div/input[3]
Form has no submit button: /html/body/div/header/div/div[1]/div[3]/form

http://www.bbc.co.uk/mundo
Field has no label or title attribute: /html/body/div/header/div/div[1]/div[3]/form/div/input[4]
Form has no submit button: /html/body/div/header/div/div[1]/div[3]/form
```

**Comment by EmmaJP** [comment]
<img width="1073" alt="screen shot 2016-01-04 at 12 09 02" src="https://cloud.githubusercontent.com/assets/7227747/12089372/fe30dfe4-b2db-11e5-9cb5-3930b112fcf7.png">

The relative element and code ... it seems the issue is the use of an image as a submit mechanism.  It should be OK (http://webaim.org/techniques/forms/controls#button), but possibly the alt should simply be the function, ie. "Search" (http://www.w3.org/TR/WCAG20-TECHS/H36.html).  This may still require the tests to be revisited.

**Comment by greenc05** [comment]
Yes I still see this error on the latest version of the tool @EmmaJP So looks like the tests need to be revisited to get this to pass please @joshski @mattwynne    

http://www.bbc.co.uk/news
- Form has no submit button: /html/body/div[1]/header/div/div[1]/div[3]/form
